src/components/legend/index.js

```diff
@@ -521,8 +555,93 @@ legend.draw = function(td) {
             });
         });
 
+    // Position and size the legend
     legend.repositionLegend(td, traces);
 
+    // Scroll section must be executed after repositionLegend.
+    // It requires the legend width, height, x and y to position the scrollbox
+    // and these values are mutated in repositionLegend.
+    var gs = fullLayout._size,
+        lx = gs.l + gs.w * opts.x,
+        ly = gs.t + gs.h * (1-opts.y);
+
+    if(opts.xanchor === 'right' || (opts.xanchor === 'auto' && opts.x >= 2 / 3)) {
+        lx -= opts.width;
+    }
+    else if(opts.xanchor === 'center' || (opts.xanchor === 'auto' && opts.x > 1 / 3)) {
+        lx -= opts.width / 2;
+    }
+
+    if(opts.yanchor === 'bottom' || (opts.yanchor === 'auto' && opts.y <= 1 / 3)) {
+        ly -= opts.height;
+    }
+    else if(opts.yanchor === 'middle' || (opts.yanchor === 'auto' && opts.y < 2 / 3)) {
+        ly -= opts.height / 2;
+    }
+
+    // Deal with scrolling
+    var plotHeight = fullLayout.height - fullLayout.margin.b,
+        scrollheight = Math.min(plotHeight - ly, opts.height),
+        scrollPosition = scrollBox.attr('data-scroll') ? scrollBox.attr('data-scroll') : 0;
+
+    bg.attr({ width: opts.width, height: scrollheight });
+    scrollBox.attr('transform', 'translate(0, ' + scrollPosition + ')');
+
+    legendsvg.call(Drawing.setRect, lx, ly, opts.width, scrollheight);
+
+    if(td.firstRender && opts.height - scrollheight > 0 && !td._context.staticPlot){
+
+        legendsvg.node().addEventListener('wheel', function(e){
+            e.preventDefault();
+            scrollHandler(e.deltaY / 20);
+        });
+
+        scrollBar.node().addEventListener('mousedown', function(e) {
+            e.preventDefault();
+
+            function mMove(e) {
+                if(e.buttons === 1){
+                    scrollHandler(e.movementY);
```

Comment:
This felt a little a crude on my laptop.

The mouse wheel scaling is very smooth though :+1: